lib/chef/client.rb

```diff
@@ -957,6 +959,12 @@ def has_admin_privileges?
 
       Chef::ReservedNames::Win32::Security.has_admin_privileges?
     end
+
+    # Register the data collector reporter to send event information to the
+    # data collector server
+    def register_data_collector_reporter
+      events.register(Chef::DataCollector::Reporter.new) if Chef::DataCollector.register_reporter?
```

Comment:
Instead of making a new method I think it makes more sense to just move this into the [`register_reporters`](https://github.com/chef/chef/blob/247bfc6c26520aa0e46332b13dc98f77d22b95e5/lib/chef/client.rb#L411) method